src/machine/machine_atsamd51.go

```diff
@@ -2297,3 +2297,60 @@ func ResetProcessor() {
 
 	arm.SystemReset()
 }
+
+// DAC on the SAMD51.
+type DAC struct {
+}
+
+var (
+	DAC0 = DAC{}
+)
+
+// DACConfig placeholder for future expansion.
+type DACConfig struct {
+}
+
+// Configure the DAC.
+// output pin must already be configured.
+func (dac DAC) Configure(config DACConfig) {
+	// Turn on clock for DAC
+	sam.MCLK.APBDMASK.SetBits(sam.MCLK_APBDMASK_DAC_)
+
+	// Use Generic Clock Generator 4 as source for DAC.
+	sam.GCLK.PCHCTRL[42].Set((sam.GCLK_PCHCTRL_GEN_GCLK4 << sam.GCLK_PCHCTRL_GEN_Pos) | sam.GCLK_PCHCTRL_CHEN)
+	for sam.GCLK.SYNCBUSY.HasBits(sam.GCLK_SYNCBUSY_GENCTRL_GCLK4 << sam.GCLK_SYNCBUSY_GENCTRL_Pos) {
+	}
+
+	// reset DAC
+	sam.DAC.CTRLA.Set(sam.DAC_CTRLA_SWRST)
+
+	// wait for reset complete
+	for sam.DAC.CTRLA.HasBits(sam.DAC_CTRLA_SWRST) {
+	}
+	for sam.DAC.SYNCBUSY.HasBits(sam.DAC_SYNCBUSY_SWRST) {
+	}
+
+	// enable
+	sam.DAC.CTRLB.Set(sam.DAC_CTRLB_REFSEL_VREFPU)
```

Comment:
When 0 is written to DATA0 while DAC_CTRLB_REFSEL_VDDANA is set, the voltage becomes VREF/2.
I think this is the same problem as the following, but I don't know what is happening.
https://www.avrfreaks.net/comment/2620686#comment-2620686

Reply:
But it works as expected with `sam.DAC_CTRLB_REFSEL_VREFPU`?

Reply:
@deadprogram 
Yes.
The voltage at 4095 is a little low, but it's almost exactly what you want.
On the other hand, VDDANA seems to work like differential mode.

![image](https://user-images.githubusercontent.com/9251039/86187269-5d0eaf80-bb76-11ea-850c-10dd6a305bf2.png)

Reply:
Well, the proof is in the signal analysis above. Good image and work on that @sago35 I am satisfied it is correct!